File: app/src/main/java/com/example/robles_farma/MyApplication.java
package com.example.robles_farma; // Asegúrate que el paquete sea el correcto

import android.app.Application;
import android.content.Context;

public class MyApplication extends Application {

    private static Context appContext;

    @Override
    public void onCreate() {
        super.onCreate();
        appContext = getApplicationContext();
    }

    /**
     * Devuelve el contexto de la aplicación,
     * disponible en cualquier parte del código.
     */
    public static Context getAppContext() {
        return appContext;
    }
}

File: app/src/main/java/com/example/robles_farma/MyApplication.kt
package com.example.robles_farma

import android.app.Application
import android.content.Context
import coil.ImageLoader
import coil.ImageLoaderFactory
import coil.decode.SvgDecoder

class MyApplication : Application(), ImageLoaderFactory {

    override fun onCreate() {
        super.onCreate()
        appContext = applicationContext
    }

    override fun newImageLoader(): ImageLoader {
        return ImageLoader.Builder(this)
            .components {
                add(SvgDecoder.Factory())
            }
            .crossfade(true)
            .build()
    }

    companion object {
        // 'lateinit' promete que inicializaremos esto antes de usarlo
        private lateinit var appContext: Context

        // ⚠️ IMPORTANTE: Esta anotación hace que Java vea este método como "static"
        @JvmStatic
        fun getAppContext(): Context {
            return appContext
        }
    }
}